torchvision/models/regnet.py

```diff
@@ -0,0 +1,785 @@
+# Modified from
+# https://github.com/facebookresearch/ClassyVision/blob/main/classy_vision/models/anynet.py
+# https://github.com/facebookresearch/ClassyVision/blob/main/classy_vision/models/regnet.py
+
+
+import numpy as np
+import math
+import torch
+
+from collections import OrderedDict
+from enum import Enum, auto
+from typing import Any, Callable, List, Optional, Tuple
+from torch import nn, Tensor
+
+from .._internally_replaced_utils import load_state_dict_from_url
+from torchvision.models.mobilenetv2 import _make_divisible
+
+model_urls = {
+    # TODO(kazhang): add pretrained weights
+    "regnet_y_400m": "",
+}
+
+
+class _SqueezeExcitation(nn.Module):
+    """
+    Squeeze and excitation layer from
+    `"Squeeze-and-Excitation Networks" <https://arxiv.org/pdf/1709.01507>`_.
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        reduction_ratio: Optional[int] = 16,
+        reduced_channels: Optional[int] = None,
+        activation: Optional[nn.Module] = None,
+    ) -> None:
+        super().__init__()
+        self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
+
+        # Either reduction_ratio is defined, or out_channels is defined,
+        # neither both nor none of them
+        assert bool(reduction_ratio) != bool(reduced_channels)
+
+        if activation is None:
+            activation = nn.ReLU()
+
+        reduced_channels = (
+            in_channels // reduction_ratio if reduced_channels is None else reduced_channels
+        )
+        self.excitation = nn.Sequential(
+            nn.Conv2d(in_channels, reduced_channels, kernel_size=1, stride=1, bias=True),
+            activation,
+            nn.Conv2d(reduced_channels, in_channels, kernel_size=1, stride=1, bias=True),
+            nn.Sigmoid(),
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        x_squeezed = self.avgpool(x)
+        x_excited = self.excitation(x_squeezed)
+        x_scaled = x * x_excited
+        return x_scaled
+
+
+class BasicTransform(nn.Sequential):
```

Comment:
> Could we reuse the ConvBNActivation layer from here?

Do you mean replacing line 79-81 with `ConvBNActivation`? or inherit from `ConvBNActivation` and add `Conv2D`, final_bn?

> passing all the layers on the constructor rather than attaching them on self.

I don't have a strong preference here. With currently implementation, we could divide layers into blocks (a, b, c), see `BottleneckTransform`. I assume the layer name would be slightly more readable when extracting features (e.g. `block4-4.f.final_bn`)

Reply:
> Do you mean replacing line 79-81 with ConvBNActivation? or inherit from ConvBNActivation and add Conv2D, final_bn?

I was thinking of replacing. That would also mean that you would be able to extend the original `ConvBNActivation` to meet your needs (in a BC manner) to make it reusable.

> With currently implementation, we could divide layers into blocks (a, b, c)

I personally prefer passing it on the constructor but that's just my preference. @fmassa could you weight in here?